foojay-resolver/src/main/kotlin/org/gradle/toolchains/foojay/distributions.kt

```diff
@@ -30,24 +30,34 @@ fun match(
         vendor: JvmVendorSpec,
         implementation: JvmImplementation,
         version: JavaLanguageVersion
-): Distribution? {
-    if (JvmImplementation.J9 == implementation) {
-        return distributions.firstOrNull { it.name == j9Aliases[vendor] }
-    }
-    if (JvmVendorSpec.GRAAL_VM == vendor) {
-        return match(distributions, JvmVendorSpec.matching("Graal VM CE " + version.asInt()))
+): List<Distribution> = when {
+    JvmImplementation.J9 == implementation -> distributions.filter { it.name == j9Aliases[vendor] }
+    JvmVendorSpec.GRAAL_VM == vendor -> match(distributions, JvmVendorSpec.matching("Graal VM CE " + version.asInt()))
+    else -> match(distributions, vendor)
+}
+
+fun match(distributions: List<Distribution>, vendor: JvmVendorSpec): List<Distribution> {
+    if (vendor == any()) {
+        return findAllDistributionsForKnownVendors(distributions)
     }
-    return match(distributions, vendor)
+
+    return findExactMatch(distributions, vendor) ?: findAllMatchingDistributions(distributions, vendor)
 }
 
-fun match(distributions: List<Distribution>, vendor: JvmVendorSpec): Distribution? {
-    val exactMatchByAlias = distributions.firstOrNull { it.name == vendorAliases[vendor] }
-    if (exactMatchByAlias != null) return exactMatchByAlias
+fun findAllDistributionsForKnownVendors(distributions: List<Distribution>): List<Distribution> =
+    distributions
+        .filter { it.name in vendorAliases.values }
+        .sortedBy { vendorAliases.values.indexOf(it.name) }
```

Comment:
This sorting also doesn't make sense. Why would the order of `vendorAliases` mean anything? We can make it so, but then we need a better way of enforcing it than an unordered map.

Reply:
Furthermore the map returned by `mapOf` is sorted.